FinniversKit/Sources/Recycling/GridViews/NeighborhoodProfile/NeighborhoodProfileView.swift

```diff
@@ -102,79 +125,45 @@ public final class NeighborhoodProfileView: UIView {
-    public override func systemLayoutSizeFitting(
-        _ targetSize: CGSize,
-        withHorizontalFittingPriority horizontalFittingPriority: UILayoutPriority,
-        verticalFittingPriority: UILayoutPriority
-    ) -> CGSize {
-        var height = NeighborhoodProfileView.headerSpacingTop
-        height += NeighborhoodProfileHeaderView.height(forTitle: viewModel.title, width: targetSize.width)
-        height += .spacingS + collectionViewHeight(forItemHeight: calculateItemSize().height)
 
-        if isPagingEnabled {
-            height += pageControl.intrinsicContentSize.height + .spacingS
-        } else {
-            height += .spacingM
+        if let banner = viewModel.banner {
+            setupBanner(with: banner)
         }
-
-        return CGSize(
-            width: targetSize.width,
-            height: height
-        )
     }
 
     // MARK: - Setup
 
     private func setup() {
         backgroundColor = .bgSecondary
 
-        addSubview(headerView)
-        addSubview(collectionView)
+        containerStackView.addArrangedSubviews([centeredContainerView(with: [headerView]), collectionView])
 
         if isPagingEnabled {
-            addSubview(pageControl)
+            containerStackView.addArrangedSubview(pageControl)
         }
 
-        var constraints = [
-            headerView.topAnchor.constraint(equalTo: topAnchor, constant: NeighborhoodProfileView.headerSpacingTop),
-            headerView.leadingAnchor.constraint(equalTo: leadingAnchor, constant: .spacingM),
-            headerView.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -.spacingM),
+        addSubview(containerStackView)
 
-            collectionView.topAnchor.constraint(equalTo: headerView.bottomAnchor, constant: .spacingS),
-            collectionView.leadingAnchor.constraint(equalTo: leadingAnchor),
-            collectionView.trailingAnchor.constraint(equalTo: trailingAnchor),
+        let constraints = [
+            containerStackView.topAnchor.constraint(equalTo: topAnchor, constant: .spacingM),
+            containerStackView.bottomAnchor.constraint(equalTo: bottomAnchor, constant: -.spacingM),
+            containerStackView.leadingAnchor.constraint(equalTo: leadingAnchor),
+            containerStackView.trailingAnchor.constraint(equalTo: trailingAnchor),
             collectionViewHeightConstraint,
         ]
 
-        if isPagingEnabled {
-            constraints.append(contentsOf: [
-                pageControl.topAnchor.constraint(equalTo: collectionView.bottomAnchor),
-                pageControl.centerXAnchor.constraint(equalTo: centerXAnchor),
-                bottomAnchor.constraint(equalTo: pageControl.bottomAnchor, constant: .spacingS)
-            ])
-        } else {
-            constraints.append(
-                bottomAnchor.constraint(equalTo: collectionView.bottomAnchor, constant: .spacingM)
-            )
-        }
-
         NSLayoutConstraint.activate(constraints)
```

Comment:
I think we don't need to store the constraints in a variable, since we use them immediately afterwards:

```swift
NSLayoutConstraint.activate([
    containerStackView.topAnchor.constraint(equalTo: topAnchor, constant: .spacingM),
    containerStackView.bottomAnchor.constraint(equalTo: bottomAnchor, constant: -.spacingM),
    containerStackView.leadingAnchor.constraint(equalTo: leadingAnchor),
    containerStackView.trailingAnchor.constraint(equalTo: trailingAnchor),
    collectionViewHeightConstraint,
])
```